packages/actions/test/unit/firebase.test.ts

```diff
@@ -72,13 +101,171 @@ describe("Firebase", () => {
+            const { userApp } = initializeUserServices()
+            it("should revert when provided the wrong credentials", async () => {
+                await expect(signInToFirebaseWithCredentials(userApp, new OAuthCredential())).to.be.rejectedWith(
+                    "Firebase: Invalid IdP response/credential: http://localhost?&providerId=undefined (auth/invalid-credential-or-provider-id)."
+                )
+            })
+            it("should revert when the application is not configured correctly", async () => {
+                expect(() => signInToFirebaseWithCredentials(initializeApp(), new OAuthCredential())).to.throw(
+                    "Firebase: Need to provide options, when not being deployed to hosting via source. (app/no-options)."
+                )
+            })
+            it("should sign in to Firebase with the provided credentials", async () => {
+                // nb. this test requires a working OAuth2 automated flow.
+            })
+            it("should sign in to Firebase with the provided credentials and return the user", async () => {
+                // nb. this test requires a working OAuth2 automated flow.
+            })
+        })
+
+        describe("initializeFirebaseCoreServices()", () => {
+            it("should successfully initialize Firebase services with correct credentials", async () => {
+                const { firebaseApp, firestoreDatabase, firebaseFunctions } = await initializeFirebaseCoreServices(
+                    process.env.FIREBASE_API_KEY!,
+                    process.env.FIREBASE_AUTH_DOMAIN!,
+                    process.env.FIREBASE_PROJECT_ID!,
+                    process.env.FIREBASE_MESSAGING_SENDER_ID!,
+                    process.env.FIREBASE_APP_ID!
+                )
+
+                expect(firebaseApp).to.not.be.null
+                expect(firestoreDatabase).to.not.be.null
+                expect(firebaseFunctions).to.not.be.null
+            })
+        })
+    }
 
     /** Helpers */
-    describe("initializeFirebaseApp()", () => {})
-    describe("getFirestoreDatabase()", () => {})
-    describe("getFirebaseFunctions()", () => {})
+    describe("initializeFirebaseApp()", () => {
+        it("should revert when the application is not configured correctly", async () => {
+            expect(() =>
+                initializeFirebaseApp({
+                    apiKey: "wrong",
+                    authDomain: "wrong",
+                    projectId: "wrong",
+                    messagingSenderId: "wrong",
+                    appId: "wrong"
+                })
+            ).to.throw(
+                "Firebase: Firebase App named '[DEFAULT]' already exists with different options or config (app/duplicate-app)."
+            )
+        })
+        if (envType === TestingEnvironment.PRODUCTION) {
+            it("should initialize a Firebase application", async () => {
+                const firebaseApp = initializeFirebaseApp({
```

Comment:
nit: create a test utility/helper to make `initializeFirebaseApp` taking as parameters `apiKey`, `authDomain`, `projectId`, `messagingSenderId` and `appId` in order to reduce code duplicates.